packages/video_player/video_player/lib/video_player.dart

```diff
@@ -378,6 +379,9 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
 
   Future<ClosedCaptionFile>? _closedCaptionFileFuture;
   ClosedCaptionFile? _closedCaptionFile;
+  ClosedCaptionFile? _previousClosedCaptionFile;
+  List<Caption>? _sortedCaptions;
```

Comment:
Looking at the `ClosedCaptionFile` type to check the above, I realized it's storing the entire contents of the file in memory, which means that adding this variable is doubling the memory usage of the captioning system.

Do you have any benchmarks about lookup time in a profile build before and after this change with non-trivial files? Given that this is a memory/speed tradeoff, rather than simply an optimization, we need to understand the actual values involved here to know if this is a net positive.